src/screens/cars/AddCar.tsx

```diff
@@ -0,0 +1,374 @@
+                onSelect={(carConverter: CarConverter) => this.setState({ selectedConverter: carConverter })}
+              />
+            )}
+          />
+          <Input
+            placeholder={Constants.VIN_FULL}
+            placeholderTextColor={commonColors.disabledDark}
+            labelStyle={style.inputLabel}
+            autoCapitalize={'none'}
+            autoCorrect={false}
+            label={`${Constants.VIN}*`}
+            errorMessage={!this.checkVIN() && vin && I18n.t('cars.invalidVIN')}
+            errorStyle={style.errorText}
+            inputStyle={style.selectDropdownRowText}
+            onChangeText={(newVin: string) => this.setState({ vin: newVin })}
+          />
+          <Input
+            placeholder={I18n.t('cars.licensePlate')}
+            placeholderTextColor={commonColors.disabledDark}
+            label={`${I18n.t('cars.licensePlate')}*`}
+            labelStyle={style.inputLabel}
+            autoCapitalize={'none'}
+            autoCorrect={false}
+            errorStyle={style.errorText}
+            errorMessage={!this.checkLicensePlate() && licensePlate && I18n.t('cars.invalidLicensePlate')}
+            inputStyle={style.selectDropdownRowText}
+            onChangeText={(newLicensePlate: string) => this.setState({ licensePlate: newLicensePlate })}
+          />
+          {this.securityProvider?.canListUsers() && (
+            <ModalSelect<User>
+              openable={true}
+              disabled={false}
+              defaultItem={selectedUser}
+              renderItem={(user: User) => <UserComponent shadowed={true} user={user} navigation={navigation} />}
+              onItemsSelected={(users: User[]) => this.setState({ selectedUser: users?.[0] })}
+              navigation={navigation}
+              selectionMode={ItemSelectionMode.SINGLE}>
+              <Users navigation={navigation} />
+            </ModalSelect>
+          )}
+          <TouchableOpacity onPress={() => this.setState({ isDefault: !isDefault })} style={style.defaultContainer}>
+            <CheckBox style={formStyles.checkbox} checked={isDefault} disabled={true} />
+            <Text style={style.text}>{I18n.t('cars.defaultCar')}</Text>
+          </TouchableOpacity>
+          <View style={style.carTypeContainer}>
+            <TouchableOpacity onPress={() => this.setState({ type: CarType.COMPANY })} style={style.typeContainer}>
+              <RadioButton.Android
+                color={commonColors.textColor}
+                uncheckedColor={commonColors.textColor}
+                onPress={() => this.setState({ type: CarType.COMPANY })}
+                value={'Company'}
+                style={style.radioButton}
+                status={type === CarType.COMPANY ? 'checked' : 'unchecked'}
+              />
+              <Text style={style.text}>{I18n.t('carTypes.companyCar')}</Text>
+            </TouchableOpacity>
+            {this.securityProvider?.isAdmin() && (
```

Comment:
isAdmin() should be avoided (check with Mahdi / Remi)
You should check projectedFields from the list.

Reply:
The projected fields would tell what types are available ?

Reply:
It would tell what fields you have the right to see.
You can align with Mahdi / Remi.

Reply:
In progres, talked to Mahdi